starboard/nplb/maximum_player_configuration_explorer.h

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_NPLB_MAXIMUM_PLAYER_CONFIGURATION_EXPLORER_H_
+#define STARBOARD_NPLB_MAXIMUM_PLAYER_CONFIGURATION_EXPLORER_H_
+
+#include <functional>
+#include <memory>
+#include <set>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "starboard/player.h"
+#include "starboard/shared/starboard/media/media_util.h"
+#include "starboard/testing/fake_graphics_context_provider.h"
+
+namespace starboard {
+namespace nplb {
+
+// The partial order relation here is: Given two vectors, a and b, a >= b if and
+// only if a[i] >= b[i] for all indices i.
+bool PosetGreaterThanOrEqualTo(const std::vector<int>& a,
+                               const std::vector<int>& b);
+
+// Given a vector v, test if it is contained within a max elements set s.
+// v is contained in a max elements set s if and only if there is a max element
+// m within s such that m >= v.
+template <typename VectorContainer>
+bool PosetContainedByMaxElementsSet(const VectorContainer& max_elements_set,
```

Comment:
VectorContainer is slightly ambiguous here, not sure if it's possible to use something like:

template <template<typename T> Container>
bool PosetContainedByMaxElementsSet(const Container<std::vector<int>>& max_elements_set,

Reply:
Replaced by:
template <template <typename val_t, typename...> class container_t>